pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -1220,41 +1220,58 @@ func Convert_v1_Firmware_To_related_apis(vmi *v1.VirtualMachineInstance, domain
 		domain.Spec.OS.KernelArgs = firmware.KernelBoot.KernelArgs
 	}
 
-	if firmware.ACPI != nil {
-		path, err := getSlicMountedPath(vmi.Spec.Volumes, firmware.ACPI.SlicNameRef)
-		if err != nil {
-			log.Log.Object(vmi).Warningf("Failed to get supported path for Volume: %s", firmware.ACPI.SlicNameRef)
-			return err
-		}
+	if err := Convert_v1_Firmware_ACPI_To_related_apis(firmware, domain, vmi.Spec.Volumes); err != nil {
+		return err
+	}
 
-		domain.Spec.OS.ACPI = &api.OSACPI{
-			Table: api.ACPITable{
-				Type: "slic",
-				Path: path,
-			},
-		}
+	return nil
+}
+
+func Convert_v1_Firmware_ACPI_To_related_apis(firmware *v1.Firmware, domain *api.Domain, volumes []v1.Volume) error {
+	if firmware.ACPI == nil {
+		return nil
+	}
+
+	if firmware.ACPI.SlicNameRef == "" && firmware.ACPI.MsdmNameRef == "" {
+		return fmt.Errorf("No ACPI tables were set. Expecting at least one.")
 	}
 
+	if domain.Spec.OS.ACPI == nil {
+		domain.Spec.OS.ACPI = &api.OSACPI{}
+	}
+
+	domain.Spec.OS.ACPI.Table = append(domain.Spec.OS.ACPI.Table,
+		createACPITable("slic", firmware.ACPI.SlicNameRef, volumes)...)
+	domain.Spec.OS.ACPI.Table = append(domain.Spec.OS.ACPI.Table,
+		createACPITable("msdm", firmware.ACPI.MsdmNameRef, volumes)...)
 	return nil
 }
 
-func getSlicMountedPath(volumes []v1.Volume, name string) (string, error) {
-	// We need to know the the volume type referred by @name
+func createACPITable(source, volumeName string, volumes []v1.Volume) []api.ACPITable {
```

Comment:
Can we have multiple tables for SLIC/MSDM? Why doesn't this return only a single table?

Reply:
You are calling `createACPITable` only with fixed name refs, so it can always match at most a single volume per table type?

Reply:
Tried to do similar to what we did with webhook validation as you [seemed](https://github.com/kubevirt/kubevirt/pull/14681#discussion_r2084546817) to like it.

There should be a 1:1 match between volume and ACPI table. The helper should find the VolumeName pointed by NameRef in volumes for the given table.

I'm always open to suggestions, if you prefer it written in a different way.